File: swift/MaxConsecutiveOnes.swift
// https://leetcode.com/problems/max-consecutive-ones/

class Solution {
    func findMaxConsecutiveOnes(_ nums: [Int]) -> Int {
        var curr = 0
        var max = 0
        for n in nums {
            if n == 1 {
                curr += 1
            } else {
                curr = 0
            }
            if curr > max {
                max = curr
            }
        }
        return max
    }
}

File: c/MaxConsecutiveOnes.c
// https://leetcode.com/problems/max-consecutive-ones/

int findMaxConsecutiveOnes(int* nums, int numsSize){
    int c = 0, m = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 1)
            c++;
        else
            c = 0;
        if (c > m)
            m = c;
    }
    return m;
}